inference-engine/src/auto_plugin/auto_exec_network.cpp

```diff
@@ -13,40 +13,85 @@
 namespace AutoPlugin {
 using namespace InferenceEngine;
 
-AutoExecutableNetwork::AutoExecutableNetwork(const SoExecutableNetworkInternal& network, bool enablePerfCount) :
-    _network(network), _enablePerfCount(enablePerfCount) {
+AutoExecutableNetwork::AutoExecutableNetwork(NetworkFuture cpuFuture,
+                                             NetworkFuture acceleratorFuture,
+                                             bool          enablePerfCount)
+                                             : _cpuFuture(std::move(cpuFuture))
+                                             , _acceleratorFuture(std::move(acceleratorFuture))
+                                             , _enablePerfCount(enablePerfCount) {
+    // we wait for any network to become ready (maybe this will already an actual device)
+    if (_cpuFuture.valid()) {
```

Comment:
seemingly the future is always valid.
waht @apankratovantonp was trying to project that the .get() below may rethrow exception if the network loading to the CPU fails. Notice that unlike actual accelerator load, we may catch this exception (as CPU is a "helper" device only).

Reply:
Thanks Maxim, well, I have used std::async() instead of Executor with tasks, it seems no failure now, all CI has passed. Also, I fixed some logic error, I believe the segment fault is solved (maybe it caused by the wrong used task/executor/future).
This PR is still missing some implementation you mentioned in email and tickets, I will continue to update it, and it is not completed yet. Of course, I will be happy if you give me any suggestion during my development, :) .  Thanks!

Reply:
seems like your std::async impl is doing something very differently, as I don't see the debug printfs in the succesful tests log...
notice that the test may also pass becuase the stad::async may run serially (as you didn't force to run it in parallel).
Finally, as the original std:therad based impl also filaed, likely the reason is not the async method itslf (whther it is therad/packed_task,executor/etc) but inaccurate capture  of e.g. 'this' (as opposite to shared_ptr) etc.

I strongly sughest to remove the capturing by value in the lambdas and carefully review what is captured and whta is lifetime of this (and if it is needed to be aligned)

Reply:
Yes, the std::async logic is different from init task impl because the original hot_swap logic is wrong.
I have force the std::async to run in parallel now.
I have changed capturing by reference, do you mean that? (you:"remove the capturing by value in the lambdas")
I think the code is ready for review now, :)